sympy/solvers/ode/tests/test_single.py

```diff
@@ -501,16 +501,14 @@ def test_nth_linear_constant_coeff_undetermined_coefficients():
 
 
 def test_nth_order_reducible():
-    from sympy.solvers.ode.ode import _nth_order_reducible_match
-
-    F = lambda eq: _nth_order_reducible_match(eq, f(x))
+    F = lambda obj: NthOrderReducible(obj)._matches()
     D = Derivative
-    assert F(D(y*f(x), x, y) + D(f(x), x)) is None
-    assert F(D(y*f(y), y, y) + D(f(y), y)) is None
-    assert F(f(x)*D(f(x), x) + D(f(x), x, 2)) is None
-    assert F(D(x*f(y), y, 2) + D(u*y*f(x), x, 3)) is None  # no simplification by design
-    assert F(D(f(y), y, 2) + D(f(y), y, 3) + D(f(x), x, 4)) is None
-    assert F(D(f(x), x, 2) + D(f(x), x, 3)) == dict(n=2)
+    assert F(SingleODEProblem(D(y*f(x), x, y) + D(f(x), x), f(x), x)) == False
+    assert F(SingleODEProblem(D(y*f(y), y, y) + D(f(y), y), f(x), x)) == False
+    assert F(SingleODEProblem(f(x)*D(f(x), x) + D(f(x), x, 2), f(x), x))== False
+    assert F(SingleODEProblem(D(x*f(y), y, 2) + D(u*y*f(x), x, 3), f(x), x)) == False  # no simplification by design
+    assert F(SingleODEProblem(D(f(y), y, 2) + D(f(y), y, 3) + D(f(x), x, 4), f(x), x)) == False
+    assert F(SingleODEProblem(D(f(x), x, 2) + D(f(x), x, 3), f(x), x)) == True
```

Comment:
Maybe just change the function `F` rather than adding the same to all examples

Reply:
Sorry, I couldn't understand the part change the function F?

Reply:
The function `F` is supposed to handle the repetitive parts in these test examples. You've just added repetitive changes to each example when you could have just changed the function `F`.

Reply:
````F = lambda eq: NthOrderReducible(SingleODEProblem(eq, f(x) ,  x) )._matches()
D = Derivative
assert F(D(y*f(x), x, y) + D(f(x), x)) == False
````
Something like this?